core_backend/app/auth/dependencies.py

```diff
@@ -140,3 +149,27 @@ def create_access_token(username: str) -> str:
     payload["type"] = "access_token"
 
     return jwt.encode(payload, JWT_SECRET, algorithm=JWT_ALGORITHM)
+
+
+async def rate_limiter(
+    request: Request,
+    user_db: UserDB = Depends(authenticate_key),
+) -> None:
+    """
+    Rate limiter for the API calls. Gets daily quota and decrement it
+    """
+    username = user_db.username
+    key = f"remaining-calls:{username}"
+    redis = request.app.state.redis
+    ttl = await redis.ttl(key)
+    # if key does not exist, set the key and value
+    if ttl == -2:
```

Comment:
So -2 means key doesn't exist? Mind using a var to make this obvious? E.g. KEY_DOES_NOT_EXIST = -2